### PR TITLE
docs: RDataFrame add multi-threaded example

### PR DESCRIPTION
**Pull request by ianna**

### CODE REVIEW

File: docs/user-guide/how-to-convert-rdataframe.md [hunk 1]
```diff
@@ -58,7 +58,7 @@ The dictionary key defines a column name in RDataFrame.
 df = ak.to_rdataframe({"x": array_x, "y": array_y, "z": array_z})
 ```
 
-The {func} `ak.to_rdataframe` function presents a generated on demand Awkward Array view as an `RDataFrame` source. There is a small overhead of generating Awkward RDataSource C++ code. This operation does not execute the `RDataFrame` event loop. The array data are not copied.
+The {func}`ak.to_rdataframe` function presents a generated on demand Awkward Array view as an `RDataFrame` source. There is a small overhead of generating Awkward RDataSource C++ code. This operation does not execute the `RDataFrame` event loop. The array data are not copied.
```
Comment (agoose77) [inline docs/user-guide/how-to-convert-rdataframe.md]:
```suggestion
The {func}`ak.to_rdataframe` function presents a generated-on-demand Awkward Array view as an `RDataFrame` source. There is a small overhead of generating Awkward RDataSource C++ code. This operation does not execute the `RDataFrame` event loop. The array data are not copied.
```

File: docs/user-guide/how-to-convert-rdataframe.md [hunk 2]
```diff
@@ -74,7 +74,7 @@ Awkward Arrays are dynamically typed, so in a C++ context, the type name is hash
 From RDataFrame to Awkward
 --------------------------
 
-The function for `RDataFrame`  → Awkward conversion is {func}`ak.from_rdataframe`. The argument to this function requires a tuple of strings that are the `RDataFrame` column names. This function always returns
+The function for `RDataFrame`  → Awkward conversion is {func}`ak.from_rdataframe`. The argument to this function requires a tuple of strings that are the `RDataFrame` column names. By default this function returns
```
Comment (agoose77) [inline docs/user-guide/how-to-convert-rdataframe.md]:
```suggestion
The function for `RDataFrame`  → Awkward conversion is {func}`ak.from_rdataframe`. The argument to this function accepts a tuple of strings that are the `RDataFrame` column names. By default this function returns
```

**Review by agoose77 — APPROVED** [review]

**Comment by agoose77** [comment]
@ianna can you reproduce the docs build failure locally? It works for me :cry:

**Comment by ianna** [comment]
> @ianna can you reproduce the docs build failure locally? It works for me 😢

nope...

**Comment by agoose77** [comment]
I think this CI is failing due to a caching issue. Not exactly sure why this is happening, but I can't reproduce it locally. I hope that merging this into main will just *work* once the caches elapse.

**Comment by agoose77** [comment]
CI _was_ failing due to caching, but not in the way I anticipated. `uproot` pulls in `awkward` from PyPI, which we then try to install manually in a subsequent step. As the versions match right now, we never get our local build. The fix is to pass `--upgrade --no-deps` to ensure this.

**Comment by agoose77** [comment]
@ianna I noticed that the ROOT test failed, which I didn't see before. I haven't touched this code, so I'm surprised that it randomly fails: https://github.com/scikit-hep/awkward/actions/runs/4298483128/jobs/7492636536

**Comment by ianna** [comment]
> @ianna I noticed that the ROOT test failed, which I didn't see before. I haven't touched this code, so I'm surprised that it randomly fails: https://github.com/scikit-hep/awkward/actions/runs/4298483128/jobs/7492636536

Sorry, it looks like I've missed to set `keep_order=True`. I'll submit a PR asap.

**Comment by ianna** [comment]
> > @ianna I noticed that the ROOT test failed, which I didn't see before. I haven't touched this code, so I'm surprised that it randomly fails: https://github.com/scikit-hep/awkward/actions/runs/4298483128/jobs/7492636536
> 
> Sorry, it looks like I've missed to set `keep_order=True`. I'll submit a PR asap.

Ok, the failing test did not expect a multi-threaded event loop. It looks like the setting is propagated from another test/docs? I'll need to check the rest of the tests and either disable multi-threading or set the parameter: https://github.com/scikit-hep/awkward/pull/2276

**Review by jpivarski — APPROVED** [review]
Very nice, thanks!

**Pull request merged**